PixArt/models/PixArt_blocks.py

```diff
@@ -14,19 +14,35 @@
 import torch.nn.functional as F
 from timm.models.vision_transformer import Mlp, Attention as Attention_
 from einops import rearrange
+from functools import reduce
+import operator # operator.mul
+
+sdpa_32b = None
+Q_4GB_LIMIT = 16000000
+"""If q is greater than this, the operation will likely require >4GB VRAM, which will fail on Intel Arc Alchemist GPUs without a workaround."""
 
 from comfy import model_management
 if model_management.xformers_enabled():
     import xformers
     import xformers.ops
 else:
-    print("""
-########################################
- PixArt: Not using xformers!
-  Expect images to be non-deterministic!
-  Batch sizes > 1 are most likely broken
-########################################
-""")
+    if model_management.xpu_available:
+        import intel_extension_for_pytorch as ipex
+        import os
+        if not torch.xpu.has_fp64_dtype() and not os.environ.get('IPEX_FORCE_ATTENTION_SLICE', None):
+            from ...utils.IPEX.attention import scaled_dot_product_attention_32_bit
+            sdpa_32b = scaled_dot_product_attention_32_bit
+            print("Using IPEX 4GB SDPA workaround")
+        else:
+            print("No IPEX 4GB workaround")
+    else: # Batch sizes > 1 are not broken with IPEX.
```

Comment:
It's not broken on torch either I just forgot to remove this warning lol. Feel free to delete it.